skopt/searchcv.py

```diff
@@ -319,6 +291,11 @@ def _check_search_space(self, search_space):
         if isinstance(search_space, dict):
             search_space = [search_space]
 
+        if len(self.search_spaces) == 0:
```

Comment:
In the rest of the method we use the `search_space` passed as argument. We should be consistent

Reply:
When does this happen? A user passes an empty list/dict?

Reply:
I'm changing it a little so that uses search_space, good catch. It will also be triggered both with an input empty dict or empty list. However it wouldn't trigger with a list of empty dicts. It looks like check_dimension (L333) would raise an error for any empty dict anyway.

This method  `check_search_space` already looks very similar to GridSearchCV's check_param_grid https://github.com/scikit-learn/scikit-learn/blob/f3320a6f1ab2d060fd49c94a6e3386291a684ec7/sklearn/model_selection/_search.py#L343 .